doc/architectural/goto-program-transformations.md

```diff
@@ -0,0 +1,311 @@
+\ingroup module_hidden
+
+\page goto-program-transformations Goto Program Transformations
+
+\section required-transforms Core Transformation Passes
+
+This section lists the transformation passes that must have been applied for the
+goto model to be in core goto format.
+
+Note that the passes are listed below in the order they are currently called in
+CBMC. While all dependencies on the ordering are not fully documented, the
```

Comment:
Can I read that as "Until all dependencies are fully documented and checked, the following order should be used."?

Reply:
You could. I think it is also important to recognise that the ordering in this documentation is well tested, in comparison to any alternative ordering of passes. Therefore the guarantees of proper functioning for alternative orders will be weaker, even if pass inter-dependencies were to be well documented.

Reply:
Sure. There is what works and what is expected to work. At some point we would expect things to work if you met the documented dependencies, and if they didn't that would be an issue to get fixed.